src/common/Header/SearchPanel/index.js

```diff
@@ -1,14 +1,42 @@
-import React from "react";
-import { useLocation } from "react-router-dom";
+import React, { useEffect, useRef } from "react";
+import { useHistory, useLocation } from "react-router-dom";
 import { toMovies } from "../../../routes";
 import { SearchInput } from "./styled";
+import { useQueryParameter } from "../../../useQueryParameters";
+import { search as searchParameterName } from "../../../queryParamNames";
 
 const SearchPanel = () => {
     const location = useLocation();
+    let query = useQueryParameter(searchParameterName);
+    let timeout = 0;
+    const history = useHistory();
+    const inputRef = useRef(null);
+
+    useEffect(() => {
+        inputRef.current.value = query;
+    }, [location]);
```

Comment:
To jest bardzo ciekawe rozwiązanie i może się mylę, ale takie w sumie niekonwencjonalne, bo z pominięciem atrybutu value inputa. Sam jestem ciekawy jak oceni to np. Krzysiek. Mój pomysł jak widzieliście był taki, żeby wartość wyszukiwana była w slice i połaczyć to z value inputa, ale ten pomysł jest na prawdę ciekawy i jest sporym uproszczeniem ;)